pandas/io/json/_json.py

```diff
@@ -718,6 +733,15 @@ def _combine_lines(self, lines) -> str:
         lines = filter(None, map(lambda x: x.strip(), lines))
         return "[" + ",".join(lines) + "]"
 
+    def _jsonstring_to_list_generaor(self, data: str) -> Iterator[str]:
+        prev_index = -1
+        while True:
+            next_index = data.find("\n", prev_index + 1)
```

Comment:
this is not a real iterator, use something like: https://stackoverflow.com/questions/4586026/splitting-a-string-into-an-iterator

Reply:
built an iterator using re.finditer